### ISSUE TITLE
GA4 feature tour can be substantially delayed

### ISSUE THREAD

**Issue by bethanylang** [google/site-kit-wp#6909]
## Bug Description

As reported by @aaemnnosttv in Bug Bash:

When the GA4 reporting feature tour is triggered, we also simultaneously enable all sections of the dashboard as if they were all in view which invokes all of their requests. This is particularly observable when clicking the CTA to view the GA4 version of the dashboard from its dedicated banner notification. This is because many requests are dispatched all at once which can delay the time it takes for the GA4 components focused in the tour to finish loading for the tour to start.

## Steps to reproduce

1. Turn on the GA4 dashboard for the first time
2. Observe slow load time of feature tour

**Update:** Note that the conditions for displaying the feature tour are changing, and the tour will no longer be triggered on page load, but rather in response to a user action - clicking a banner CTA. See related discussion here: https://github.com/google/site-kit-wp/issues/6914#issuecomment-1517649485. This presents the opportunity to preload reports in order to reduce the delay of the feature tour appearing.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The perceived delay in the GA4 Reporting feature tour appearing should be significantly reduced

## Implementation Brief

The approach taken here is to preload the entire widgets.

Within assets/js/googlesitekit/widgets/datastore/widgets.js:

- Update `registerWidget()` to allow an optional `isPreloaded()` callback to be passed as part of settings (similar to the existing `isActive()` callback).
  - Ensure this is passed through as part of the payload.
- Update the filtering in `getWidgets()` so that when `isActive()` is present and returns false, and the `isPreloaded()` callback is defined, an additional check is made to `widget.isPreloaded( select )`, and if this returns true the widget is still deemed active.
- Add an `isWidgetPreloaded()` selector which takes a widget slug and returns the result of `isPreloaded( select )` for that widget if it exists.

Within assets/js/googlesitekit/widgets/components/WidgetRenderer.js:

- Use the new `isWidgetPreloaded()` selector to check for the widget's preloaded status.
- If it's preloaded, wrap the returned `widgetElement` in a div which has the `HIDDEN_CLASS` applied.

Within assets/js/modules/analytics/index.js and assets/js/modules/search-console/index.js:

- For each `analytics-4` widget, add an `isPreloaded( select )` callback to the definition that returns the result of the `shouldPromptGA4DashboardView()` selector.

### Test Coverage

- Add test coverage for all of the datastore changes, and the update to `WidgetRenderer`.

## QA Brief

- Setup Site Kit with the `ga4Reporting` feature flag enabled.
- Setup Analytics with both UA and GA4 connected.
- Switch the Dashboard View back to Universal Analytics.
- As we are now dismissing the feature tour behind the scenes when setting up Site Kit with GA4, we need to remove the dismissed state in order to be able to trigger it again.
  - Delete the user's `wp_googlesitekitpersistent_dismissed_tours` meta field.
- Clear session storage to ensure the various reports are not cached.
- Navigate to the Dashboard, and click on the **Update dashboard** CTA when it appears (allow a couple of seconds for reading the notice, while the GA4 Search Funnel widget preloads in the background).
- The GA4 Reporting feature tour should appear with a significantly reduced delay.

**Follow-up detail**

The follow-up change here post-initial QA makes it so that steps 2 and 3 of the tour will always be available regardless of how quickly they are navigated to.

To test, simply invoke the tour with a clean cache and try going to the second step right away.
You should see that the step is always navigable and that the rows of the table will remain in a loading state while it finishes loading – although this will only be possible while the GA4 reporting tour is active (otherwise it will be more of a full-widget loading state as today).

## Changelog entry

* Fix bug that caused the GA4 Dashboard Feature Tour to delay appearing.

**Comment by aaemnnosttv** [comment]
One idea perhaps for an IB here from the original review https://github.com/google/site-kit-wp/pull/6827#discussion_r1160091745

The concern about blocking other tours while checking requirements is valid so if we did go this route, we'd want to make sure that it wouldn't be blocked by the widget not being in view.

**Comment by aaemnnosttv** [comment]
Thanks for the breakdown @techanvil. First, shaving 500ms off by not loading the entire dashboard is still a 20% improvement, so that sounds like a worthwhile improvement (if not too complicated), especially as I would imagine this could be more substantial (at least in terms of total time) on lower bandwidth connections.

Regarding the widget itself, if there isn't a straightforward improvement we can make to reduce the delay of the tour starting then it probably isn't worth sinking extra time into right now and we can come back to this as a nice-to-have before launch if we have time.

One idea that comes to mind would be to trigger/start some of the GA4 report requests from the banner notification when switching the dashboard version as a way to prioritize them, but this would only make a difference in that specific scenario and not other cases where someone could simply end up on the GA4 dashboard for the first time.

Another idea might be to create a different kind of feature tour in a modal which wouldn't depend on data from the API and would be something we could show right away. This could also allow the dashboard to continue loading in the background so that the user could see this tour immediately and everything else would be more/less loaded by the time the tour was completed/dismissed. This would be a more substantial change I think though and sounds a bit ambitious to accomplish before launch.

**Comment by techanvil** [comment]
Thanks @aaemnnosttv. One thing to point out, the approximated time reduction for not loading the entire dashboard is only ~250ms, so only a ~10% improvement (these figures are of course _very_ rough).

I think it should be fairly straightforward to implement though, so could still be worth doing. The only potential complication is that there'd be more chance of the user trying to navigating to the next tour step while the step target is not yet in the DOM. If we accept that chance, it's easy, if we want to mitigate it then it involves more work (e.g. disabling the "Next" button until the target is present).

> One idea that comes to mind would be to trigger/start some of the GA4 report requests from the banner notification when switching the dashboard version as a way to prioritize them, but this would only make a difference in that specific scenario and not other cases where someone could simply end up on the GA4 dashboard for the first time.

This could indeed be effective although as you've pointed out, only in the scenario where the banner is present. It would be nice though, in that situation, given how the feature tour will effectively appear in response to a user action, so it would make that interaction more snappy. But, it's debatable whether we want to spend the engineering effort on this one pre-launch, could be somewhat complex for a relatively low impact feature.

> Another idea might be to create a different kind of feature tour in a modal which wouldn't depend on data from the API and would be something we could show right away. This could also allow the dashboard to continue loading in the background so that the user could see this tour immediately and everything else would be more/less loaded by the time the tour was completed/dismissed. This would be a more substantial change I think though and sounds a bit ambitious to accomplish before launch.

Yup, this could also work although it's really starting to veer into uncharted territory which would require some Product and UX input, certainly doesn't feel like a viable pre-launch change to make.

**Comment by aaemnnosttv** [comment]
> One thing to point out, the approximated time reduction for not loading the entire dashboard is only ~250ms, so only a ~10% improvement (these figures are of course _very_ rough).

Surely loading everything on the dashboard at once with a cold cache would take longer than 2.5s, at least I feel it has in my experience. In particular, invoking requests for PageSpeed reports at the same time as all the others would make sense to avoid (I'm not counting how long these take to finish 😄). Maybe we can be a little more selective in the sections of the dashboard that we turn on in this case rather than everything since we at least know which widgets the tour step targets are in?

**Comment by techanvil** [comment]
Sorry if it was a bit unclear, but by talking about "loading everything on the dashboard" here I was actually referring to the act of toggling `forceInView` on, causing the whole dashboard to _start_ loading, not actually waiting for the whole thing to load.

> Maybe we can be a little more selective in the sections of the dashboard that we turn on in this case rather than everything since we at least know which widgets the tour step targets are in?

We could try that - it would mean forcing the just the Traffic and Content sections to be in view and leaving Speed and Monetization alone. However, it strikes me this may have negligible or even no perceivable effect depending on what's causing the current slowdown when we're toggling `forceInView`. It could be down to added contention on the GA4 API, which would still happen when toggling just the GA4 widgets on...

**Comment by aaemnnosttv** [comment]
Thanks @techanvil – going back to your initial point here 

> The bulk of the delay is due to the amount of time the `SearchFunnelWidgetGA4` widget spends in the loading state.

It looks like the component pulls 4 reports aside from those used for gathering data. In my experience this doesn't take that long to load on its own.

What do you feel would provide the best bang for our buck here to at least get it out of P0 territory?

**Comment by techanvil** [comment]
Thanks @aaemnnosttv, on further consideration I think the thing to do here is to focus on preloading the reports.

I've done a little more testing and, when toggling to GA4 Dashboard View from a cold cache (for the GA4 widgets), in this instance the time `SearchFunnelWidgetGA4` spent in the loading state was ~2.5s. I then tried toggling again with the reports cached and it dropped to ~500ms.

Bearing in mind we'll only be triggering the feature tour in response to a user action (as per #6914), this preloading should be enough to mitigate the slowness for most use cases.

I think we can specify the AC to be that the reports should be preloaded, and leave it to the IB to determine precisely how - it might be easiest to preload the entire widget(s) rather than trying to extract/replicate individual reports to preload.

I've updated the AC, and will leave this in ACR for you to take a look at.

**Comment by aaemnnosttv** [comment]
Thanks @techanvil – that sounds good to me, just a few thoughts

> * This should be achieved by preloading the reports for the GA4 widgets.

Does this mean to say we want to preload reports for all GA4 widgets or only those relevant for the tour? Should we maybe also prioritize the preloading such that the first step is unblocked as soon as possible?

This actually reminds me of another idea I had which would make the initialization of the tour available almost immediately which would be to add a new (additional) first step targeting an element that would always be there. Given we've added an indicator on the dashboard to show the "active version" (UA/GA4) perhaps we should highlight this in the first step to provide a bit more time for reports to load in the background and unblock the next step(s)?

> * These reports should only be preloaded when a CTA to start the feature tour is present.

Does this mean we'd preload the reports as soon as we display the CTA? (even before clicking that CTA?)

**Comment by aaemnnosttv** [comment]
SGTM, thanks @techanvil 👍

**Comment by techanvil** [comment]
Hey @eugene-manuilov, while I think this is a nice idea in principle, we do still have a small amount of layout shift when the widgets transition from the loading to the loaded state, as can be seen in this video clip (the viewport is 783px which is the minimum width for showing the feature tour). I think this would be problematic for the feature tour if it we took this approach.

https://user-images.githubusercontent.com/18395600/234901734-cc987eb6-c107-4bee-a5a9-441e6984d27f.mp4

**Comment by eugene-manuilov** [comment]
The layout shift is still the problem for all feature tours in our plugin, right? If I recall correctly, we should have a ticket for it. So this should be a separate issue.

The purpose of this ticket is to show the feature tour as early as possible, which can be easily achieved if we update the preview layout of two widgets.

**Comment by techanvil** [comment]
Hey @eugene-manuilov, I'm sending this back over to you for a couple of reasons.

One, although your point about layout shifts being a problem for feature tours does ring a bell, I can't actually see any open issues for it. Maybe I'm missing something, or maybe you're thinking of an issue that we've previously addressed?

Secondly, even though we may have/had some similar issues, I think this would be a bit different in nature - for one thing, the existing/previous issues would have been things we only noticed after we'd implemented them, and for another, your proposal would still be the first time we'd actually be trying to show a feature tour for widgets which are deliberately still in a loading state.

Bear in mind, although I posted that video at a narrow viewport to make the shift more obvious, it still occurs at wider viewports. Here it is at 1280px wide, where the Search Traffic widget (which is the first target for the feature tour) can be seen to shift once the All Traffic widget has loaded.

Considering the tour will be shown in the scenario where the GA4 dashboard view is enabled for the first time, this means the GA4 reports will almost certainly _not_ be cached, the feature tour will always target the initial loading-state layout alignment, and the layout will most likely shift while the user is reading the first feature item, which means the highlighted area will move and be misaligned. This will look like a glitch and be apparent for most/all users. It just doesn't seem like a good idea to me to deliberately take this approach.

Interested to hear your thoughts.

https://user-images.githubusercontent.com/18395600/235125886-dcaeaebd-76b4-41e9-90d4-ccfd1f8aebaf.mp4

**Comment by eugene-manuilov** [comment]
> One, although your point about layout shifts being a problem for feature tours does ring a bell, I can't actually see any open issues for it. Maybe I'm missing something, or maybe you're thinking of an issue that we've previously addressed?

> Bear in mind, although I posted that video at a narrow viewport to make the shift more obvious, it still occurs at wider viewports. Here it is at 1280px wide, where the Search Traffic widget (which is the first target for the feature tour) can be seen to shift once the All Traffic widget has loaded.

I don't remember whether we created such a ticket or just discussed creating one. If you didn't find it, then we probably would have just discussed creating a ticket for it. Anyway this is not a big problem and we can fix it by triggering a fake resize event when we receive a report response. Something like this: `global.dispatchEvent(new Event('resize'));`. This will signal to joyride that it needs to re-position the tooltip and everything will look okay.

> Considering the tour will be shown in the scenario where the GA4 dashboard view is enabled for the first time, this means the GA4 reports will almost certainly _not_ be cached, the feature tour will always target the initial loading-state layout alignment, and the layout will most likely shift while the user is reading the first feature item, which means the highlighted area will move and be misaligned. This will look like a glitch and be apparent for most/all users. It just doesn't seem like a good idea to me to deliberately take this approach.

Once again, this can be easily mitigated, if we tell the joyride library to re-position the tooltip when data is loaded. I'll defer to make the final decision to @aaemnnosttv. If he thinks that the suggested IB is okay to go, then I am okay with it as well.

**Comment by aaemnnosttv** [comment]
@techanvil if the main concern here is about the layout shift, this is something we should address anyways as that's not really supposed to happen and it looks like we have 2 separate layout shifts that are happening in the all traffic widget as can be seen in your screencast on the smaller viewport and only 1 on the larger. FWIW, we also don't show feature tours at all in viewports below desktop due to problems we had with Joyride in mobile so this may be somewhat less of a concern.

The tour is primarily about new/different metrics themselves rather than the values so I think it makes sense that the tour doesn't need to be blocked by everything being loaded (although of course ideally the data would be there) so I think the change that @eugene-manuilov has described would be sufficient for moving this forward with a simpler solution.

I do think the layout shift is important to fix though, but that should get its own issue and we can include the resizing in the tour here if needed as a temporary workaround.

@techanvil your suggested solution is an interesting one though and has given me some other ideas as well. I don't think these are mutually exclusive though, so let's start with the smaller change and see about addressing the layout shift and we can see about further enhancing this (and other tours) in the future 👍

**Comment by eugene-manuilov** [comment]
Ok, IB is updated.

**Comment by eugene-manuilov** [comment]
Fair point, Tom. I should have provided more details. IB is updated 👍.

**Comment by techanvil** [comment]
Thanks @eugene-manuilov, that looks great.

IB :white_check_mark:

**Comment by techanvil** [comment]
Hey @eugene-manuilov, unfortunately during development I've run into an issue with this approach.

The problem is, we don't actually know which Conversions feature tour target we'll be showing - the data block or the CTA - until _after_ we've loaded the report to retrieve the conversions (see [here](https://github.com/google/site-kit-wp/blob/e127d92e5b185f7d0a4281f32746295ec1f6f4e8/assets/js/modules/search-console/components/dashboard/SearchFunnelWidgetGA4/Overview/index.js#L199-L208)), which effectively means we need to wait until the widget has loaded before we can show the tour (we could add more logic so we show the Conversions cell as soon as that report is loaded rather than waiting for all of the widget's reports, but unless we add yet more complexity there's no guarantee that will resolve any faster than any of the other reports).

![image](https://user-images.githubusercontent.com/18395600/236250778-d051d0a1-66ac-46b7-abb4-228ff1cd200a.png)

![image](https://user-images.githubusercontent.com/18395600/236251153-004c6032-01b6-4b0b-b6d5-4e992d598784.png)

If we try to show the data block before we finish loading and then discover that we actually need to show the CTA, the feature tour ends up like this:

![image](https://user-images.githubusercontent.com/18395600/236252993-550231c1-1f59-4e3d-8d54-9aea915681fa.png)

Also, by initially showing the loading data block and then switching to the CTA, the UI looks a bit inconsistent. I've tried dispatching a resize event upon load, so at least the tour highlight updates, but it breaks the tooltip position and the way the content changes doesn't look so good either.

https://user-images.githubusercontent.com/18395600/236257568-2a34bcba-7e67-4fe2-966b-6b1d56e63cba.mp4

Taking all of this on board, I'm not sure it's worth pursuing this route after all, and suspect we may need to revert to the widget preloading approach that I had initially specced out, or take some yet-to-be-determined alternative approach.

What do you think?

**Comment by eugene-manuilov** [comment]
> The problem is, we don't actually know which Conversions feature tour target we'll be showing - the data block or the CTA - until _after_ we've loaded the report to retrieve the conversions...

Oh... this is a good point, @techanvil. I forgot about it when suggested this approach.

> ...which effectively means we need to wait until the widget has loaded before we can show the tour (we could add more logic so we show the Conversions cell as soon as that report is loaded rather than waiting for all of the widget's reports...

Yes, i think we need to adjust our logic in that component to account for these details. I think what we can do is to show the current loading state until we get information from the conversions report. As soon as we get it, we determine whether we want to display the new preloading state (if any other report is still loading) or keep showing the same current loading state if we don't have conversions and need to show the CTA.

> but unless we add yet more complexity there's no guarantee that will resolve any faster than any of the other reports

Yes, there is no guarantee for that, but it will definitely resolve not longer than if we wait for all widgets to be preloaded.

> Taking all of this on board, I'm not sure it's worth pursuing this route after all, and suspect we may need to revert to the widget preloading approach that I had initially specced out, or take some yet-to-be-determined alternative approach.

I still think we can go with this approach. The only thing that will change is that we will change the loading state from very "obscure" (the current loading state) to less "obscure" (the new loading state with titles shown) based on the status of the conversion report request. For users it will be a kind of "progressive enhancement" of the UI of that widget.

What do you think?

**Comment by eugene-manuilov** [comment]
Also, we need to determine what we should do if the user has no conversions yet when we need to show the feature tour. We can't highlight the CTA block in the feature tour because it makes no sense IMO. I think we can either don't show the feature tour at all if the user has no conversions, or show only two other steps that point to the popular pages widget titles.

@marrrmarrr @aaemnnosttv @techanvil what do you think?

**Comment by eugene-manuilov** [comment]
@techanvil 

> I do think this misses the point a bit.

Ah.. yes, my mistake, in theory if we do what you have suggested, we could potentially get all reports loaded faster by increasing complexity of widget API and useInView/useInViewSelect hooks that also would need to be updated if we want to implement preloading using your approach (you forgot to mention it in your IB) because currently rendering a widget (even with the hidden class) doesn't mean that its data will be preloaded if the widget is not in the view.

> where it initially looked conceptually simpler than the widget preloading, I now think it's at least comparable or beyond that in terms of complexity.

I think this is very debatable what is easier or more complex: changing widgets API and useInView/useInViewSelect hooks, + add new selectors or just figure out how to preload conversions and update two components.

> By still waiting for the widget to appear in a loading state, and then waiting for the report to load to determine the conversions data, we're not going to be making a dramatic reduction in the delay.

Okay, okay, how about preloading conversions in the banner by calling the `getConversionEvents` selector? We use (or used to use) this approach in a couple of places to preload accounts and properties in some components knowing that this information might be needed down the road. By doing this we will have conversions resolved by the time we need to load the widget and we won't have problems with the new preloading state. What do you think?

**Comment by techanvil** [comment]
Hey @eugene-manuilov, I have tried your suggested approach of calling `getConversionEvents()`. However, I discovered that in fact it's not just this request that's needed to determine which Conversions cell will be displayed. It also relies on an item of data in the main overview report. The loading of this report itself depends on a number of other bits of state, and extracting these for preloading the report results in a more significant refactor.

Also, for some reason the end result is still noticably slower than the preloading widgets approach. I'm seeing about 1 second from CTA click to the initial feature tour rendering for the preload requests approach, vs about 600ms for preloading widgets.

To your point about the in-view aspect - I don't think we actually need to worry about this, for the specific scenario we are addressing here, as with the CTA being in a notification banner at the top, and the widgets we're loading being in the Traffic section, they are always in-view, so I think we could take the approach of preloading widgets without modifying the in-view logic to address this in time for launch, and then update it to work for not-in-view widgets either post-launch or as and when we actually need it.

This being the case, based on the discovery so far it definitely feels like the preloading widgets approach is the less complex of the two for our current requirements, and it's more performant (obviously we could look into this aspect further, but I am a bit hesitant to spend yet more time on it considering our launch target).

In order to illustrate both the timing and relative complexity of both approaches I've created a draft PR and screencast for each of them. Please take a look, and see what you think.

**Preload requests ([PR](https://github.com/google/site-kit-wp/pull/7022/files)):**

https://github.com/google/site-kit-wp/assets/18395600/cf35b363-61db-422c-b875-7b59790eae70

**Preload widgets ([PR](https://github.com/google/site-kit-wp/pull/7023/files)):**

https://github.com/google/site-kit-wp/assets/18395600/17bae62f-0fec-46c4-b92b-28ea99f1a38e

**Comment by eugene-manuilov** [comment]
> I'm seeing about 1 second from CTA click to the initial feature tour rendering for the preload requests approach, vs about 600ms for preloading widgets.

IMO that's okay if it is being loaded 1s because users won't see the difference between 600ms and 1s anyway. @aaemnnosttv what do you think?

**Comment by techanvil** [comment]
> > I'm seeing about 1 second from CTA click to the initial feature tour rendering for the preload requests approach, vs about 600ms for preloading widgets.
> 
> IMO that's okay if it is being loaded in ~1s because users won't notice the difference between 600ms and 1s anyway. @aaemnnosttv what do you think?

Thanks @eugene-manuilov - maybe that's the case, although to be honest the 600ms feels a lot snappier. But there's also the complexity aspect to consider. If you compare the two PRs, ignore the logging, there's a _lot_ more going on in the "preload requests" one. I'm also interested to get @aaemnnosttv's take on this too.

**Comment by aaemnnosttv** [comment]
Thanks @techanvil for the side-by-side comparison with POC PRs, that's really helpful.

After reviewing both approaches, let's go with the widget preloading for now as the simpler change in terms of the implementation. Essentially both approaches preload the requests, but allowing widgets to preload has the benefit of not needing to leak details of what data it uses to do so.

Regarding the difference in timing between the two – with the difference being ~.5sec, I don't feel it's significant enough to justify choosing one approach or the other based on that alone (with both being substantially quicker than it is now of course), and it's likely that this will vary quite a bit as well, so I'm not sure we can even conclude concretely that one is faster than the other without much more testing, but I don't feel like this issue is worth that level of precision. Overall, the experience seems sufficiently enhanced IMO so let's wrap this up and revisit in a follow-up issue(s) later where we feel there is room for improvement, as this will likely come up again in the future.

Thanks @eugene-manuilov for your perspective and ideas as always!

As for next steps, @techanvil please update the IB to very briefly summarize the approach and then the main thing I think we'd want to do is add some test coverage for the changes to the Widgets API. Assigning back to you 👍

**Comment by techanvil** [comment]
Thanks @aaemnnosttv. Good high level point to make about not leaking details, that had occurred to me too at some point but I got a bit into the weeds with it all. Also on reflection, that does make sense that we can't be entirely sure one approach will always perform faster than the other without more substantial testing.

Anyway, I've updated the IB and will proceed to adding some test coverage, cheers.

**Comment by mohitwp** [comment]
### QA Update ✅ 

- Tested on dev environment.
- Compared featured tour display time on both latest and dev environment.
- On dev environment, feature tour is appearing very fast as compare to the latest environment.
- Verified feature tour trigger functionality when user clicks on 'Update to Ga4 dashboard' button and when user clicks on 'Learn what's new' button.

<details>

On latest -


https://github.com/google/site-kit-wp/assets/94359491/a3a09c5a-1b8f-4345-8f21-b298788876e6


On dev - 


https://github.com/google/site-kit-wp/assets/94359491/a879192e-fd9f-41fc-a561-2b3af0bb9591


https://github.com/google/site-kit-wp/assets/94359491/55022d1c-20f2-4853-b55e-29c4ba77e4a4

</details>

**Comment by aaemnnosttv** [comment]
### ❌ Approval

In light of #7031, (and aside from PR #7034 follow-up to #6558), there was still the potential for the tour to crash when navigating to the 2nd step if loading hadn't completed there. While this isn't explicitly part of the AC here, it isn't worth addressing in a separate issue.

I've opened a small follow-up PR to address this here by modifying the popular pages widget to be available as soon as the GA4R tour is active (regardless of the loading state). I looked at extending this to the search funnel as well but that would require more substantial changes that aren't necessary for now.

**Comment by mohitwp** [comment]
@wpdarren Are you able to generate banner where we get the button to 'Update dashboard'. Previously I was able to but now I'm not able to generate banner using the QA steps.

**Comment by techanvil** [comment]
@mohitwp, I think you may now also need to delete the user's meta field `wp_googlesitekitpersistent_dismissed_items` to ensure the notification displays, as described in https://github.com/google/site-kit-wp/issues/6932.
